pkg/sink/kafka/metrics_collector.go

```diff
@@ -15,211 +15,140 @@ package kafka
+	RxBytes int `json:"rx_bytes"`
+	// Number of topics in the metadata cache.
+	MetadataCacheCnt int `json:"metadata_cache_cnt"`
+
+	Brokers map[string]broker `json:"brokers"`
+}
 
-type saramaMetricsCollector struct {
-	changefeedID common.ChangeFeedID
-	// adminClient is used to get broker infos from broker.
-	adminClient ClusterAdminClient
-	brokers     map[int32]struct{}
-	registry    metrics.Registry
+type broker struct {
+	Name     string `json:"name"`
+	Nodeid   int    `json:"nodeid"` // -1 for bootstraps
+	Nodename string `json:"nodename"`
+	State    string `json:"state"`
+	Rtt      window `json:"rtt"`
 }
 
-func (m *saramaMetricsCollector) Run(ctx context.Context) {
-	// Initialize brokers.
-	m.updateBrokers(ctx)
+type window struct {
+	Min int `json:"min"`
+	Max int `json:"max"`
+	Avg int `json:"avg"`
+	P99 int `json:"p99"`
+}
 
-	refreshMetricsTicker := time.NewTicker(RefreshMetricsInterval)
-	refreshClusterMetaTicker := time.NewTicker(refreshClusterMetaInterval)
-	defer func() {
-		refreshMetricsTicker.Stop()
-		refreshClusterMetaTicker.Stop()
-		m.cleanupMetrics()
-	}()
+type metricsCollector struct {
+	changefeedID common.ChangeFeedID
+	config       *kafka.ConfigMap
+}
+
+// NewMetricsCollector return a kafka metrics collector based on library.
+func NewMetricsCollector(
+	changefeedID common.ChangeFeedID,
+	config *kafka.ConfigMap,
+) MetricsCollector {
+	return &metricsCollector{changefeedID: changefeedID, config: config}
+}
 
+func (m *metricsCollector) Run(ctx context.Context) {
+	_ = m.config.SetKey("statistics.interval.ms", int(RefreshMetricsInterval.Milliseconds()))
+	p, err := kafka.NewProducer(m.config)
+	if err != nil {
+		log.Error("create producer failed", zap.Error(err))
+		return
+	}
 	for {
 		select {
 		case <-ctx.Done():
-			log.Info("Kafka metrics collector stopped",
-				zap.String("namespace", m.changefeedID.Namespace()),
-				zap.String("changefeed", m.changefeedID.Name()))
+			log.Warn("Kafka metrics collector stopped",
+				zap.String("namespace", m.changefeedID.String()))
+			p.Close()
+			m.cleanupMetrics()
 			return
-		case <-refreshMetricsTicker.C:
-			m.collectBrokerMetrics()
-			m.collectProducerMetrics()
-		case <-refreshClusterMetaTicker.C:
-			m.updateBrokers(ctx)
+		case event := <-p.Events():
+			switch e := event.(type) {
+			case *kafka.Stats:
+				m.collect(e.String())
+			}
 		}
 	}
 }
 
-func (m *saramaMetricsCollector) updateBrokers(ctx context.Context) {
-	start := time.Now()
-	brokers, err := m.adminClient.GetAllBrokers(ctx)
-	if err != nil {
-		log.Warn("Get Kafka brokers failed, "+
-			"use historical brokers to collect kafka broker level metrics",
-			zap.String("namespace", m.changefeedID.Namespace()),
-			zap.String("changefeed", m.changefeedID.Name()),
-			zap.Duration("duration", time.Since(start)),
-			zap.Error(err))
+func (m *metricsCollector) collect(data string) {
+	var statistics kafkaMetrics
+	if err := json.Unmarshal([]byte(data), &statistics); err != nil {
```

Comment:
make sure recorded metrics matches the monitoring

Reply:
Yes, you can find it in https://github.com/confluentinc/librdkafka/blob/master/STATISTICS.md